### PR TITLE
20454 - Adding hide-footer attribute

### PR DESCRIPTION
**Pull request by dotpointer**
Adding hide-footer attribute.

Testing: Try to set the `hide-footer` on `cosmoz-omnitable`, the footer should be hidden.

Needed in #20454.

### CODE REVIEW

File: cosmoz-omnitable.html [hunk 1]
```diff
@@ -134,7 +134,7 @@ <h3 class="groupRow-label">
 					</cosmoz-grouped-list>
 				</div>
 			</div>
-			<div class="footer">
+			<div class$="footer [[ _getFooterHiddenClass(hideFooter) ]]">
```
Comment (nomego) [inline cosmoz-omnitable.html]:
```suggestion
			<div class$="footer [[ ifElse(hideFooter, 'hidden', '') ]]">
```

File: cosmoz-omnitable.js [hunk 2]
```diff
@@ -182,6 +182,13 @@
 				value: 0
 			},
 
+			/**
+			 * Footer visibility setting.
+			 */
+			hideFooter: {
+				type: Boolean
```
Comment (nomego) [inline cosmoz-omnitable.js]:
```suggestion
				type: Boolean,
				value: false
```

File: cosmoz-omnitable.js [hunk 3]
```diff
@@ -182,6 +182,13 @@
 				value: 0
 			},
 
+			/**
+			 * Footer visibility setting.
+			 */
+			hideFooter: {
```
Comment (nomego) [inline cosmoz-omnitable.js]: Make it routine to also add test

File: cosmoz-omnitable.js (hunk 2 above)
Comment (dotpointer) [reply cosmoz-omnitable.js]: Changed to it.

File: cosmoz-omnitable.html (hunk 1 above)
Comment (dotpointer) [reply cosmoz-omnitable.html]: Changed to it, console complains: method `ifElse` not defined when testing on frontend. Maybe local problem, re-running component update.

Comment (dotpointer) [reply cosmoz-omnitable.html]: Problem persists, ifElse is not defined.

Comment (dotpointer) [reply cosmoz-omnitable.html]: Trying to use `cosmoz-behaviors`, since it contains `ifElse`.

Comment (nomego) [reply cosmoz-omnitable.html]: Yeah sorry it seems omnitable itself doesn't use the TemplateHelperBehavior but the component depends on it. But some columns do. So yeah, please add the behavior to cosmoz-omnitable.js

Comment (dotpointer) [reply cosmoz-omnitable.html]: Seems to work with it.

File: cosmoz-omnitable.js (hunk 3 above)
Comment (dotpointer) [reply cosmoz-omnitable.js]: Test is now added, thanks to @megheaiulian for guidance.

**Comment by dotpointer** [comment]
I found that the bottom-bar were not shown if this attribute was used. I have made an update to the pull request. Also tried adding checks for footer-controls and footer-tableStats, but these failed in one of the Chrome browsers on the second run.

**Comment by dotpointer** [comment]
@nomego The need for it is because of the demo made 2018-12-11, where a request to remove the footer was made. My intent was to hide the visible footer but not the details needed to display the bottom actions. Since this never reached the demo then this could be closed if the demo should not be put into production.

**Comment by dotpointer** [comment]
Demo is over, closing this.

**Pull request closed without merging**